### ISSUE TITLE
Not able to programatically specify what file extensions are included

### ISSUE THREAD

**Issue by matthewdunsdon** [benbria/browserify-transform-tools#16]
When using browserify-transform-tools (v 1.5.1), I am not able to programatically specify what file extensions to include (``includeExtensions``).

Partial code sample:
```javascript
var DEFAULT_EXTENSIONS = ['.js']
var options = { includeExtensions: DEFAULT_EXTENSIONS }
var myStringTransform = transformTools.makeStringTransform("hello", options, function(content, transformOptions, done) {
    done(null, 'hello ' + content);
});
browserify({ entries: ['index.html'] })
    .transform(myStringTransform, { appliesTo: { includeExtensions: ['.html'] } })
    .bundle()
```

**Comment by matthewdunsdon** [comment]
Having a quick look at the code, the follow change may have caused the problem:

https://github.com/benbria/browserify-transform-tools/commit/76829faa8d65bef694ea2141e15128f150dac7c1#diff-a801f5af448ba06bc6096727b57fa60eL58

**Comment by matthewdunsdon** [comment]
I have not used coffeescript on a regular basis and am not sure if this is the correct place to make the fix, therefore I have created a reference commit (rather than a PR) showing sample fix and proposed test changes:

https://github.com/matthewdunsdon/browserify-transform-tools/commit/ec9e7b0f2fac3

**Comment by jwalton** [comment]
Thanks for tracking this down so quickly!  I'll have a look at your fix
tomorrow morning.

**Comment by jwalton** [comment]
Published as 1.5.2.  Thanks @matthewdunsdon!  I used [almost exactly your fix](https://github.com/benbria/browserify-transform-tools/commit/295e5b44d0e0722d079fbc96038fe981d9a94f71).

**Comment by matthewdunsdon** [comment]
Looking at this futher, I am not sure this quite solves all the edge cases. I have created PR #17 to address this.
